pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -1197,6 +1294,82 @@ func (nc *Controller) handleDisruption(zoneToNodeConditions map[string][]*v1.Nod
 	}
 }
 
+func (nc *Controller) podUpdated(oldPod, newPod *v1.Pod) {
+	if newPod == nil {
+		return
+	}
+	if len(newPod.Spec.NodeName) != 0 && (oldPod == nil || newPod.Spec.NodeName != oldPod.Spec.NodeName) {
+		podItem := podUpdateItem{newPod.Namespace, newPod.Name}
+		nc.podUpdateQueue.Add(podItem)
+	}
+}
+
+func (nc *Controller) doPodProcessingWorker() {
+	for {
+		obj, shutdown := nc.podUpdateQueue.Get()
+		// "podUpdateQueue" will be shutdown when "stopCh" closed;
+		// we do not need to re-check "stopCh" again.
+		if shutdown {
+			return
+		}
+
+		podItem := obj.(podUpdateItem)
+		nc.processPod(podItem)
+	}
+}
+
+func (nc *Controller) processPod(podItem podUpdateItem) {
+	defer nc.podUpdateQueue.Done(podItem)
+	pod, err := nc.podLister.Pods(podItem.namespace).Get(podItem.name)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			// If the pod was deleted, there is no need to requeue.
+			return
+		}
+		utilruntime.HandleError(fmt.Errorf("Failed to read pod %v/%v: %v", podItem.namespace, podItem.name, err))
+		nc.podUpdateQueue.Add(podItem)
+		return
+	}
+
+	nodeName := pod.Spec.NodeName
+
+	nodeHealth := nc.nodeHealthMap.getDeepCopy(nodeName)
+	if nodeHealth == nil {
+		// Pod will be handled by standard eviction pass,
+		// when node data is collected.
```

Comment:
Or the node was deleted in the meantime?

I think this comment should describe both cases.

Reply:
Done.